lib/ansible/modules/windows/win_shortcut.ps1

```diff
@@ -5,33 +5,53 @@
 
 # Based on: http://powershellblogger.com/2016/01/create-shortcuts-lnk-or-url-files-with-powershell/
 
-#Requires -Module Ansible.ModuleUtils.Legacy
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+$spec = @{
+    options = @{
+        src = @{ type='str' }
+        dest = @{ type='path'; required=$true }
+        state = @{ type='str'; default='present'; choices=@( 'absent', 'present' ) }
+        args = @{ type='str' }
+        directory = @{ type='path' }
+        hotkey = @{ type='str' }
+        icon = @{ type='path' }
+        description = @{ type='str' }
+        windowstyle = @{ type='str'; choices=@( 'maximized', 'minimized', 'normal' ) }
+    }
+    supports_check_mode = $true
+}
 
-$ErrorActionPreference = "Stop"
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
 
-$params = Parse-Args -arguments $args -supports_check_mode $true
-$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+$src = $module.Params.src
+$dest = $module.Params.dest
+$state = $module.Params.state
+$args = $module.Params.args
```

Comment:
This is dangerous, `$args` is a global variable set by PowerShell to contain the input arguments to the script/function.

Reply:
Right, we did it before as well. I guess because we don't rely on the original `$args` before.
I'll fix it.